website/docs/reference/resource-configs/contract.md

```diff
@@ -11,78 +11,90 @@ id: "contract"
 
-Tests serve as a **detective** measure against bad data quality _after_ the dbt model is (re)built.
+```yml
+models:
+  - name: dim_customers
+    config:
+      contract:
+        enforced: true
+    columns:
+      - name: customer_id
+        data_type: int
+        constraints:
+          - type: not_null
+      - name: customer_name
+        data_type: string
+```
 
-Constraints are great when you define `constraints: ['not null']` for a column in your model's schema because it'll prevent `null` values from being inserted into that column at dbt model creation time and prevent other unintended values from being inserted into that column without dbt's intervention as it relies on the database to enforce the constraint. This can **replace** the `not_null` test. However, performance issues may arise depending on your database.
+</File>
 
-Tests should be used in addition to and instead of constraints when you want to test things like `accepted_values` and `relationships`. These are usually not enforced with built-in database functionality and are not possible with constraints. Also, custom tests will allow more flexibility and address nuanced data quality issues that may not be possible with constraints.
+<File name='models/dim_customers.yml'>
 
-## Current Limitations
+Let's say your model is defined as:
+```sql
+select
+  'abc123' as customer_id,
+  'My Best Customer' as customer_name
+```
 
-- `contract` (a.k.a. `constraints_enabled`) must be configured in the yaml [`config`] property _only_. Setting this configuration via in-file config or in `dbt_project.yml` is not supported.
-- `contract` (a.k.a. `constraints_enabled`) is supported only for a SQL model materialized as `table`.
-- Prerequisite checks include the column `name,` but not yet their `data_type`. We intend to support `data_type` verification in an upcoming beta prerelease.
-- The order of columns in your `yml` file must match the order of columns returned by your model's SQL query.
-- While most data platforms support `not_null` checks, support for [additional `constraints`](resource-properties/constraints) varies by data platform.
+</File>
 
+When you `dbt run` your model, _before_ dbt has materialized it as a table in the database, you will see this error:
 ```txt
 # example error message
-Compilation Error in model constraints_example (models/constraints_examples/constraints_example.sql)
-  Please ensure the name, order, and number of columns in your `yml` file match the columns in your SQL file.
-  Schema File Columns: ['id', 'date_day', 'color']
-  SQL File Columns: ['id', 'color', 'date_day'] 
+Compilation Error in model dim_customers (models/dim_customers.sql)
+  Contracts are enabled for this model. Please ensure the name, data_type, and number of columns in your `yml` file match the columns in your SQL file.
+  Schema File Columns: customer_id INT, customer_name TEXT
+  SQL File Columns: customer_id TEXT, customer_name TEXT
 ```
 
-## Example
+## Support
```

Comment:
This is a more-positive spin on "Limitations" :) but if you think that's a clearer way of expressing it, we can switch out one for the other (or for something else)

Reply:
It looks good. Always more pleasing to the reader to put a positive spin on anything. "Don't apologize for keeping them waiting; thank them for their patience" type of experience.